### ISSUE TITLE
Colon rule doesn't show warning for generic types

### ISSUE THREAD

**Issue by commscheck** [realm/SwiftLint#2628]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

'Colon' rule does not trigger for class and struct definitions if the type is generic.

##### Complete output when running SwiftLint, including the stack trace and command used

Input:

```swift
import Foundation

print("Hello, World!")

protocol MyProtocol { }

class Baz<String> : MyProtocol { }
struct Qux<String> : MyProtocol { }
```

Command Line:

```bash
$ swiftlint lint
Linting Swift files at paths
Linting 'main.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
    * 0.30.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
    * Homebrew
* Paste your configuration file:
    * N/A

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
    * No
* Which Xcode version are you using (check `xcode-select -p`)?
    * Xcode 10.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules` to quickly test if your example is really demonstrating the issue. If your example is more complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// These trigger a warning as expected:
class Foo : MyProtocol { }
struct Bar : MyProtocol { }

// These do not:
class Baz<String> : MyProtocol { }
struct Qux<String> : MyProtocol { }
```
